napari/layers/shapes/_shapes_utils.py

```diff
@@ -699,13 +809,138 @@ def _sign_cross(x, y):
+    components except the one with the largest extent.
+    """
+    v, e = _normalize_vertices_and_edges(path, close=True)
+    n_edge = len(e)
+    if n_edge == 0:
+        # if there's not enough edges in the path, return fast.
+        # Note: this condition can probably be expanded: the smallest hole
+        # is 3 edges within 3 edges (a triangle within a triangle), so we
+        # should be able to use this exit with n_edge < 6. However, there may
+        # be some edge cases to consider when drawing partial polygons
+        return path
+    m = np.max(e)
+    csr = sparse.coo_matrix(
+        (np.ones(n_edge), tuple(e.T)), shape=(m + 1, m + 1)
+    ).tocsr()
+    n_comp, labels = sparse.csgraph.connected_components(csr, directed=False)
+    comp2idxs = _indices(labels)
+    idxs = [_get_idxs(comp2idxs, i) for i in range(n_comp)]
+    # having found the connected components, we make sure we sort the vertices
+    # in adjacent sequence.
+    # TODO: Check whether we can simplify this step.
+    #  This *might* be unnecessary work because the unique vertices might be
+    #  in the correct order anyway.
+    path_orders = [
+        sparse.csgraph.depth_first_order(
+            csr, idx[0], directed=False, return_predecessors=False
+        )
+        for idx in idxs
+    ]
+    paths = [v[path_order] for path_order in path_orders]
+    if not keep_holes:
+        # any holes will necessarily have a smaller span than the enclosing
+        # polygon, so it is sufficient to check the ptp (max-min) of each array
+        # along an arbitrary axis
+        paths = [max(paths, key=lambda p: np.ptp(p[:, 0]))]
+    return paths
+
+
+def _combine_meshes(meshes_list):
+    """Combine a list of (centers, offsets, triangles) meshes into one.
+
+    Meshes are generated as tuples of (centers, offsets, triangles), where the
+    triangles index into the centers and offsets arrays (see
+    :func:`generate_2D_edge_meshes`). To convert multiple instances into a
+    single mesh, the centers and offsets can simply be concatenated, but the
+    triangle indices must be incremented by the cumulative length of the
+    previous centers arrays.
+
+    Parameters
+    ----------
+    meshes_list : list of tuple of arrays
+        A list where each element is a tuple of (centers, offsets, triangles).
+        Centers is an Mx2 or Mx3 array, offsets is an Mx2 or Mx3 array (where
+        M is the number of vertices in the triangulation), and triangles is a
+        Px3 array of integers, where each integer is an index into the centers
+        array.
+
+    Returns
+    -------
+    centers : np.ndarray of float
+        Mx2 or Mx3 array of vertex coordinates.
+    offsets : np.ndarray of float
+        Mx2 or Mx3 array of offsets for edge width.
+    triangles : np.ndarray of int
+        Px3 array of indices into centers to form the triangles.
+    """
+    centers_list, offsets_list, triangles_list = list(zip(*meshes_list))
```

Comment:
If len on the list is 1, we should just return it without calling concat.

Reply:
Done!

Reply:
Where? Did you not pushed some changes?